File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    public int MaxDepth(TreeNode root) {
        // If the root is null
        if (root == null) {
            // Return zero
            return 0;
        }

        // Do a recursive solution to figure out the Max depth
        return 1 + Math.Max(MaxDepth(root.left), MaxDepth(root.right));
    }
}

File: Data Structures & Algorithms/subtree-of-a-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:   
    def isSubtree(self, root: Optional[TreeNode], subRoot: Optional[TreeNode]) -> bool:

        def isSameTree(s, t):
            if not s and not t:
                return True

            if not s or not t or s.val != t.val:
                return False

            return isSameTree(s.left, t.left) and isSameTree(s.right, t.right)

        def dfs(node):
            if not node:
                return False

            if isSameTree(node, subRoot):
                return True

            return dfs(node.left) or dfs(node.right)

        return dfs(root)